tools/netqtop.c

```diff
@@ -95,19 +95,24 @@ TRACEPOINT_PROBE(net, net_dev_start_xmit){
 }
 
 TRACEPOINT_PROBE(net, netif_receive_skb){
-    struct sk_buff* skb = (struct sk_buff*)args->skbaddr;
-    if(!name_filter(skb)){
+    struct sk_buff skb;
+
+    bpf_probe_read(&skb, sizeof(skb), args->skbaddr);
+    if(!name_filter(&skb)){
         return 0;
     }
 
-    u16 qid = skb->queue_mapping;
+    u16 qid = 0;
+    if (skb_rx_queue_recorded(&skb))
+        qid = skb_get_rx_queue(&skb);
```

Comment:
From my understanding, 
```
static inline u16 skb_get_rx_queue(const struct sk_buff *skb)
{
        return skb->queue_mapping - 1;
} 
```

Here qid = 0 could mean two things: there is no mapping or actual qid of 0. Do we need to differentiate this? For example, if `!skb_rx_queue_recorded(&skb)`, simply return from the future as it is not in any queue? Or if not in any queue, it means queue 0?

Reply:
- If the NIC does **not** support multi-queue feature, there is only one queue(qid is always 0).
- And if the NIC supports multi-queue feature, there are several queues with different qid(from 0 to n-1).

So I guess current logic could represent the two cases. And if we want to know weather a NIC supports multi-queue feature, we can use `ethtool -l NICNAME` to test it.

Reply:
What is the return value for skb_rx_queue_recorded(&skb) for single-queue feature? If this is false, then current code should work fine. Just adding some comments to explain is good enough.

Reply:
OK, added some comments and force-pushed!